src/com/oltpbenchmark/benchmarks/tpcc/JsonMetricsHelper.java

```diff
@@ -0,0 +1,232 @@
+import com.oltpbenchmark.util.LatencyMetricsUtil;
+import com.oltpbenchmark.util.FileUtil;
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.DecimalFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class JsonMetricsHelper {
+    private static final Logger LOG = Logger.getLogger(JsonMetricsHelper.class);
+    private TpccRunResults tpccRunResults;
+
+    public JsonMetricsHelper() {
+        tpccRunResults = new TpccRunResults();
+    }
+
+    public void setTestConfig(int numNodes, int numWH, int numDBConn, int warmuptime, int runtime, int numRetries) {
+        if(tpccRunResults.TestConfiguration == null)
+            tpccRunResults.TestConfiguration = tpccRunResults.new TestConf();
+        tpccRunResults.TestConfiguration.numNodes = numNodes;
+        tpccRunResults.TestConfiguration.numWarehouses = numWH;
+        tpccRunResults.TestConfiguration.numDBConnections = numDBConn;
+        tpccRunResults.TestConfiguration.warmupTimeInSecs = warmuptime;
+        tpccRunResults.TestConfiguration.runTimeInSecs = runtime;
+        tpccRunResults.TestConfiguration.numRetries = numRetries;
+        tpccRunResults.TestConfiguration.testStartTime = new SimpleDateFormat("dd-MM-yy_HH:mm:ss").format(new Date());
+    }
+
+    public void setTestResults(String tpmc, String efficiency, String throughput) {
+        if(tpccRunResults.Results == null)
+            tpccRunResults.Results = tpccRunResults.new RunResults();
+        tpccRunResults.Results.tpmc = Double.parseDouble(tpmc);
+        tpccRunResults.Results.efficiency = Double.parseDouble(efficiency);
+        tpccRunResults.Results.throughput = Double.parseDouble(throughput);
+    }
+
+    public void addLatency(String op, List<Integer> latencyList, List<Integer> connLatencyList) {
+        tpccRunResults.Latencies.add(getValueList(op, latencyList, connLatencyList));
+    }
+
+    public void addFailureLatency(String op, List<Integer> latencyList, List<Integer> connLatencyList) {
+        tpccRunResults.FailureLatencies.add(getValueList(op, latencyList, connLatencyList));
+    }
+
+    public void addWorkerTaskLatency(String op, String task, List<Integer> latencyList) {
+        if (!tpccRunResults.WorkerTaskLatency.containsKey(op)) {
+            tpccRunResults.WorkerTaskLatency.put(op, new ArrayList<>());
+        }
+        tpccRunResults.WorkerTaskLatency.get(op).add(getValueList(task, latencyList, null));
+    }
+
+    public void addRetry(String op, int count,int[] retryOpList) {
+        TpccRunResults.RetryAttemptsData retryObj = tpccRunResults.new RetryAttemptsData();
+        retryObj.count = count;
+        retryObj.retriesFailureCount = Arrays.asList(retryOpList);
+        tpccRunResults.RetryAttempts.put(op,retryObj);
+    }
+
+    private TpccRunResults.LatencyList getValueList(String op, List<Integer> latencyList,
```

Comment:
rename `getLatencyList` or something more corresponding to what is being returned